lib/addons/ebs-csi-driver/index.ts

```diff
@@ -15,44 +17,121 @@ const versionMap: Map<KubernetesVersion, string> = new Map([
 /**
  * Interface for EBS CSI Driver EKS add-on options
  */
-export interface EbsCsiDriverAddOnProps {
-    /**
-     * Version of the driver to deploy
-     */
-    version?: string;
-    /**
-     * List of KMS keys to be used for encryption
-     */
-    kmsKeys?: kms.Key[];
+export interface EbsCsiDriverAddOnProps extends CoreAddOnProps {
+  /**
+   * List of KMS keys to be used for encryption
+   */
+  kmsKeys?: kms.Key[];
+  /**
+   * StorageClass to be used for the addon
+   */
+  storageClass?: string;
 }
 
 /**
  * Default values for the add-on
  */
-const defaultProps = {
-    addOnName: "aws-ebs-csi-driver",
-    version: "auto",
-    versionMap: versionMap,
-    saName: "ebs-csi-controller-sa",
+const defaultProps: CoreAddOnProps & EbsCsiDriverAddOnProps = {
+  addOnName: "aws-ebs-csi-driver",
+  version: "v1.23.0-eksbuild.1",
+  saName: "ebs-csi-controller-sa",
+  storageClass: "gp3", // Set the default StorageClass to gp3

```

Comment:
@nrajb you removed versionMap from the default props, it has impact on how this addon will behave during build time. 
please restore, it will also take care of the warning above stating the versionMap is not used.

Reply:
corrected

Reply:
@nrjab please don't resolve conversation, it should be done by the reviewer, otherwise the comments are hidden.